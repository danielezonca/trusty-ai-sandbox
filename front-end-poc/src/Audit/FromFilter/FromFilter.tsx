import React, { useCallback } from "react";
import DatePicker from "../../Shared/components/DatePicker/DatePicker";

type FromFilterProps = {
  fromDate: string;
  maxDate: string;
  onFromDateUpdate: (date: string) => void;
};

const FromFilter = (props: FromFilterProps) => {
  const { fromDate, maxDate, onFromDateUpdate } = props;
  const updateFromDate = useCallback((date: string) => onFromDateUpdate(date), [onFromDateUpdate]);

  return (
    <DatePicker
      fromDate={fromDate}
      maxDate={maxDate}
      value={fromDate}
      label="Starting date"
      id="audit-from-date"
      onDateUpdate={updateFromDate}
    />
  );
};

export default FromFilter;
