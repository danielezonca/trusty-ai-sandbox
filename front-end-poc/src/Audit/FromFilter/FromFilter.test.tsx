import React from "react";
import FromFilter from "./FromFilter";
import { mount, shallow } from "enzyme";

const defaultProps = {
  fromDate: "2020-01-01",
  maxDate: "2020-02-29",
  onFromDateUpdate: jest.fn(),
};

describe("From filter", () => {
  test("renders", () => {
    const wrapper = shallow(<FromFilter {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("handles date update", () => {
    const wrapper = mount(<FromFilter {...defaultProps} />);
    const fromDate = "2020-01-20";
    wrapper.props().onFromDateUpdate(fromDate);

    expect(defaultProps.onFromDateUpdate).toHaveBeenCalledTimes(1);
    expect(defaultProps.onFromDateUpdate).toHaveBeenCalledWith(fromDate);
  });
});
