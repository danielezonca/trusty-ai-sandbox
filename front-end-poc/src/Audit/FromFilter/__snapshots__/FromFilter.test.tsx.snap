// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From filter renders 1`] = `
<DatePicker
  fromDate="2020-01-01"
  id="audit-from-date"
  label="Starting date"
  maxDate="2020-02-29"
  onDateUpdate={[Function]}
  value="2020-01-01"
/>
`;
